Source/Widgets/Cesium3DTilesInspector/Cesium3DTilesInspectorViewModel.js

```diff
@@ -0,0 +1,880 @@
+                } else {
+                    that._eventHandler.removeInputAction(ScreenSpaceEventType.LEFT_CLICK);
+                }
+            }
+        });
+        /**
+         * Gets or sets the pick state
+         *
+         * @type {Boolean}
+         * @default false
+         */
+        this.pickActive = false;
+
+        this._style = undefined;
+        this._shouldStyle = false;
+        this._tileset = undefined;
+        this._feature = undefined;
+        this._definedProperties = ['propertiesText', 'dynamicSSE', 'colorBlendMode', 'picking', 'colorize', 'wireframe', 'showBoundingVolumes', 'showContentBoundingVolumes',
+                                   'showRequestVolumes', 'showGeometricError', 'freezeFrame', 'maximumSSE', 'dynamicSSEDensity', 'dynamicSSEFactor', 'pickActive'];
+        this._removePostRenderEvent = scene.postRender.addEventListener(function() {
+            that._update();
+        });
+    }
+
+    defineProperties(Cesium3DTilesInspectorViewModel.prototype, {
+        /**
+         * Gets the stats text.  This property is observable.
+         * @memberof Cesium3DTilesInspectorViewModel.prototype
+         * @type {String}
+         * @readonly
+         */
+        statsText : {
+            get : function() {
+                return this._statsText;
+            }
+        },
+        /**
+         * Gets the pick stats text.  This property is observable.
+         * @memberof Cesium3DTilesInspectorViewModel.prototype
+         * @type {String}
+         * @readonly
+         */
+        pickStatsText : {
+            get : function() {
+                return this._pickStatsText;
+            }
+        },
+
+        /**
+         * Gets the available blend modes
+         * @memberof Cesium3DTilesInspectorViewModel.prototype
+         * @type {Object[]}
+         * @readonly
+         */
+        colorBlendModes : {
+            get : function() {
+                return colorBlendModes;
+            }
+        },
+
+        /**
+         * Gets the editor error message
+         * @memberof Cesium3DTilesInspectorViewModel.prototype
+         * @type {String}
+         * @readonly
+         */
+        editorError : {
+            get : function() {
+                return this._editorError;
+            }
+        },
+
+        /**
+         * Gets or sets the tileset of the view model
+         * @memberof Cesium3DTilesInspectorViewModel.prototype
+         * @type {Cesium3DTileset}
+         */
+        tileset : {
+            get : function() {
+                return this._tileset;
+            },
+
+            set : function(tileset) {
+                this._tileset = tileset;
+                this._style = undefined;
+                this.feature = undefined;
+
+                if (defined(tileset)) {
+                    var that = this;
+                    tileset.readyPromise.then(function(t) {
+                        that._properties(t.properties);
```

Comment:
This is a race condition.  If the viewModel is destroyed before  the tileset is ready, this will throw.  Check `that.isDestroyed()` first.